### PR TITLE
feat: add hashing checks on dataset creation

### PR DESCRIPTION
**Pull request by mwlodarzc**

### CODE REVIEW

**Comment by mwlodarzc** [comment]
Checks for hashes in dataset
Changed to config path usage instead on config in train.py and scripts/dataset.py
Waiting for a review

File: train_corrupted.py [hunk 1]
```diff
@@ -95,8 +95,7 @@ def train(config_path, workdir):
     initial_step = int(state['step'])
 
     # Build data iterators
-    trainloader, testloader = datasets.get_dataset(
-        config, uniform_dequantization=config.data.uniform_dequantization)
+    trainloader, testloader = datasets.get_dataset(config_path)
```
Comment (ggruszczynski) [inline train_corrupted.py:98]: config it is already loaded
`config = load_config_from_path(config_path)`

File: train_corrupted.py [hunk 2]
```diff
@@ -132,7 +131,7 @@ def train(config_path, workdir):
     delta = config.model.sigma*1.25
     n_denoising_steps = config.solver.n_denoising_steps   
     initial_sample = sampling.get_initial_corrupted_sample(
-        config, n_denoising_steps, corruptor)
+        config_path, n_denoising_steps, corruptor)
```
Comment (ggruszczynski) [inline train_corrupted.py:134]: config it is already loaded
`config = load_config_from_path(config_path)`

File: train.py [hunk 3]
```diff
@@ -68,8 +70,7 @@ def train(config, workdir):
     initial_step = int(state['step'])
 
     # Build data iterators
-    trainloader, testloader = datasets.get_dataset(
-        config, uniform_dequantization=config.data.uniform_dequantization)
+    trainloader, testloader = datasets.get_dataset(config_path)
```
Comment (ggruszczynski) [inline train.py:73]: config it is already loaded
`config = load_config_from_path(config_path)`

File: train.py [hunk 4]
```diff
@@ -89,7 +90,7 @@ def train(config, workdir):
 
     # Building sampling functions
     delta = config.model.sigma*1.25 # std deviation of the sampling noise
-    initial_sample, _ = sampling.get_initial_sample(config, heat_forward_module, delta)
+    initial_sample, _ = sampling.get_initial_sample(config_path, heat_forward_module, delta)
```
Comment (ggruszczynski) [inline train.py:93]: config it is already loaded
`config = load_config_from_path(config_path)`

File: scripts/sampling.py [hunk 5]
```diff
@@ -150,24 +151,20 @@ def sampler(model):
...
 
-def get_initial_sample(config, forward_heat_module, delta, batch_size=None):
+def get_initial_sample(config_path, forward_heat_module, delta, batch_size=None):
     """Take a draw from the prior p(u_K)"""
-    trainloader, _ = datasets.get_dataset(config,
-                                          uniform_dequantization=config.data.uniform_dequantization,
-                                          train_batch_size=batch_size)
-
+    trainloader, _ = datasets.get_dataset(config_path, train_batch_size=batch_size)
+    config = load_config_from_path(config_path)
```
Comment (ggruszczynski) [inline scripts/sampling.py:157]: please pass parsed config

config it is already loaded
`config = load_config_from_path(config_path)`

File: scripts/sampling.py [hunk 6]
```diff
@@ -150,24 +151,20 @@ def sampler(model):
...
     initial_sample = next(iter(trainloader))[0].to(config.device)
     original_images = initial_sample.clone()
     initial_sample = forward_heat_module(initial_sample, config.model.K * torch.ones(initial_sample.shape[0], dtype=torch.long).to(config.device))
     return initial_sample, original_images
 
 
-def get_initial_corrupted_sample(dataset_config, corruption_amount, solver: BaseCorruptor, batch_size=None):
+def get_initial_corrupted_sample(dataset_config_path, corruption_amount, solver: BaseCorruptor, batch_size=None):
```
Comment (ggruszczynski) [inline scripts/sampling.py:164]: config it is already loaded
`config = load_config_from_path(config_path)`

File: scripts/datasets.py [hunk 7]
```diff
@@ -37,20 +41,21 @@ def get_dataset(config, uniform_dequantization=False, train_batch_size=None,
     https://github.com/fyu/lsun
     """
 
+    config = load_config_from_path(config_path)
```
Comment (ggruszczynski) [inline scripts/datasets.py:44]: config it is already loaded
`config = load_config_from_path(config_path)`

**Review by ggruszczynski — APPROVED** [review]
LGTM (looks good to me)

**Pull request merged**